assets/js/src/core/modules/data-object/hooks/use-data-object-helper.ts

```diff
@@ -121,5 +128,37 @@ export const useDataObjectHelper = (): UseDataObjectReturn => {
     }
   }
 
-  return { openDataObject, executeDataObjectTask }
+  const formatPath = async (items: IFormatPathItems[], fieldName: string, dataObjectId: number): Promise<DataObjectFormatPathApiResponse> => {
+    const targets = items.reduce((acc, item) => {
+      acc[`object_${item.id}`] = {
+        id: item.id,
+        type: item.type,
+        label: item.fullPath,
+        path: item.fullPath,
+        nicePathKey: `object_${item.id}`
+      }
+      return acc
+    }, {})
+
+    const { data } = await store.dispatch(api.endpoints.dataObjectFormatPath.initiate({
+      body: {
+        objectId: dataObjectId,
+        targets,
+        context: {
+          containerType: 'object',
+          fieldname: fieldName,
+          objectId: dataObjectId,
+          layoutId: '0'
+        }
+      }
+    }))
+
+    if (data === undefined) {
+      throw new Error('Failed to format path: API response is undefined.')
```

Comment:
Please add the `trackError(new ApiError(...))` logic here if RTK query returns an error in the request.